lib/db/set.go

```diff
@@ -184,7 +189,20 @@ func (s *FileSet) Drop(device protocol.DeviceID) {
 		s.meta.resetAll(device)
 	}
 
-	if err := s.meta.toDB(s.db, []byte(s.folder)); backend.IsClosed(err) {
+	t, err := s.db.newReadWriteTransaction()
+	if backend.IsClosed(err) {
+		return
+	} else if err != nil {
+		panic(err)
+	}
+	defer t.close()
+
+	if err := s.meta.toDB(t, []byte(s.folder)); backend.IsClosed(err) {
+		return
+	} else if err != nil {
+		panic(err)
+	}
+	if err := t.Commit(); backend.IsClosed(err) {
```

Comment:
This is to protect against writing our half the keys on shutdown? Feels like metadata might be better of as a single key, then we would not need trasactions.

Reply:
It's so that toDB can take a transaction and be part of the same batch as the files it accounts, instead of just being hammered as a separate write later.

Reply:
It is a single key, fwiw.

Reply:
But in this specific case the transaction is not used for anything else, so why can't we just do straight writes?

Reply:
Because toDB takes a transaction, as it must. Or we could duplicate it into toDBWithoutTransaction, but ... we don't drop devices that often.